File: src/main/java/br/com/dusty/dkits/listener/gameplay/EntityDamageByEntityListener.java
package br.com.dusty.dkits.listener.gameplay;

import br.com.dusty.dkits.gamer.Gamer;
import org.bukkit.entity.Player;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.entity.EntityDamageByEntityEvent;

public class EntityDamageByEntityListener implements Listener {
	
	@EventHandler
	public void onEntityDamage(EntityDamageByEntityEvent event) {
		if(event.getEntity() instanceof Player && event.getDamager() instanceof Player){
			Gamer gamer = Gamer.of((Player) event.getEntity());
			Gamer damager = Gamer.of((Player) event.getDamager());
			
			gamer.setCombatTag(10000);
			gamer.setCombatPartner(damager);
			
			damager.setCombatTag(10000);
			damager.setCombatPartner(gamer);
		}
	}
}

File: src/main/java/br/com/dusty/dkits/listener/gameplay/EntityDamageByEntityListener.kt
package br.com.dusty.dkits.listener.gameplay

import br.com.dusty.dkits.gamer.Gamer
import org.bukkit.entity.Player
import org.bukkit.event.EventHandler
import org.bukkit.event.Listener
import org.bukkit.event.entity.EntityDamageByEntityEvent

class EntityDamageByEntityListener: Listener {

	@EventHandler
	fun onEntityDamage(event: EntityDamageByEntityEvent) {
		if (event.entity is Player && event.damager is Player) {
			val gamer = Gamer.of(event.entity as Player)
			val damager = Gamer.of(event.damager as Player)

			gamer.setCombatTag(10000)
			gamer.combatPartner = damager

			damager.setCombatTag(10000)
			damager.combatPartner = gamer
		}
	}
}
